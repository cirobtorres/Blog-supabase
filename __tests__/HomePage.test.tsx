// import "@testing-library/jest-dom";
// import { render, screen, waitFor } from "@testing-library/react";
// import { faker } from "@faker-js/faker";
// import { createMockArticles } from "@/__mocks__/mockArticles";
// import { createBrowserAppClient } from "../supabase/client";
// import HomePage from "../app/page";

// faker.seed(1); // Snapshots consistency

// const mockArticlesUserOff = createMockArticles();
// const mockArticlesUserOn = createMockArticles({ isAuthorNull: false });

// jest.mock("next/headers", () => ({
//   cookies: jest.fn(() => ({
//     setAll: () => [],
//     getAll: () => [],
//   })),
// }));

// jest.mock("@/supabase/client", () => ({
//   createBrowserAppClient: jest.fn(),
// }));

// jest.mock("@/supabase/server", () => ({
//   createServerAppClient: jest.fn(() => ({
//     auth: {
//       getUser: jest.fn(() => ({
//         data: { user: { user_metadata: { displayName: "UserTest" } } },
//       })),
//     },
//   })),
// }));

// describe("HomePage", () => {
//   it("renders in the document (user off)", async () => {
//     // Returns a select and overrideTypes mocking.
//     const mockFrom = {
//       select: jest.fn().mockReturnValue({
//         order: jest.fn().mockReturnValue({
//           overrideTypes: jest.fn().mockReturnValue({
//             data: null,
//             error: { message: "Error" },
//           }),
//         }),
//       }),
//     };

//     (createBrowserAppClient as jest.Mock).mockReturnValue({
//       from: jest.fn().mockReturnValue(mockFrom),
//     });

//     render(await HomePage());

//     await waitFor(() => {
//       mockArticlesUserOff.forEach((articleMock) => {
//         const article = screen.getByText(articleMock.title);
//         expect(article).toBeInTheDocument();
//       });
//     });
//   });

//   it("renders in the document (user on)", async () => {
//     // Returns a select and overrideTypes mocking.
//     const mockFrom = {
//       select: jest.fn().mockReturnValue({
//         overrideTypes: jest.fn().mockReturnValue({
//           data: mockArticlesUserOn,
//           error: null,
//         }),
//       }),
//     };

//     (createBrowserAppClient as jest.Mock).mockReturnValue({
//       from: jest.fn().mockReturnValue(mockFrom),
//     });

//     render(await HomePage());

//     await waitFor(() => {
//       mockArticlesUserOn.forEach((articleMock) => {
//         const article = screen.getByText(articleMock.title);
//         expect(article).toBeInTheDocument();
//       });
//     });
//   });

//   it("renders error when there is an error to the supabase query", async () => {
//     const consoleErrorSpy = jest
//       .spyOn(console, "error")
//       .mockImplementation(() => {});

//     // Returns a select and overrideTypes mocking.
//     const mockFrom = {
//       select: jest.fn().mockReturnValue({
//         overrideTypes: jest.fn().mockReturnValue({
//           data: null,
//           error: { message: "Error" },
//         }),
//       }),
//     };

//     (createBrowserAppClient as jest.Mock).mockReturnValue({
//       from: jest.fn().mockReturnValue(mockFrom),
//     });

//     render(await HomePage());

//     expect(
//       screen.getByText("Wops, an error has occurred.")
//     ).toBeInTheDocument();

//     expect(consoleErrorSpy).toHaveBeenCalled();

//     consoleErrorSpy.mockRestore();
//   });

//   it("renders 'No article' when there is no article", async () => {
//     // Returns a select and overrideTypes mocking.
//     const mockFrom = {
//       select: jest.fn().mockReturnValue({
//         overrideTypes: jest.fn().mockReturnValue({
//           data: [],
//           error: null,
//         }),
//       }),
//     };

//     (createBrowserAppClient as jest.Mock).mockReturnValue({
//       from: jest.fn().mockReturnValue(mockFrom),
//     });

//     render(await HomePage());

//     expect(screen.getByText("No article yet =|")).toBeInTheDocument();
//   });

//   it("matches the snapshot", async () => {
//     // Returns a select and overrideTypes mocking.
//     const mockFrom = {
//       select: jest.fn().mockReturnValue({
//         overrideTypes: jest.fn().mockReturnValue({
//           data: mockArticlesUserOn,
//           error: null,
//         }),
//       }),
//     };

//     (createBrowserAppClient as jest.Mock).mockReturnValue({
//       from: jest.fn().mockReturnValue(mockFrom),
//     });

//     const { asFragment } = render(await HomePage());

//     expect(asFragment()).toMatchSnapshot();
//   });
// });
