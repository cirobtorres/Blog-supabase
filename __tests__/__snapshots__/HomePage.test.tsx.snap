// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage matches the snapshot 1`] = `
<DocumentFragment>
  <main
    class="max-w-7xl mx-auto min-h-screen bg-slate-900"
  >
    <section
      class="grid grid-rows-[32px_100px_1fr] py-10 mx-10 bg-slate-800"
    >
      <div
        class="flex justify-between mx-4 bg-slate-950"
      >
        <div
          class="flex items-center gap-2"
        >
          <a
            class="w-fit py-1 px-2 bg-slate-700"
            href="/articles"
          >
            Articles
          </a>
        </div>
        <div
          class="flex items-center gap-2"
        >
          <p>
            UserTest
          </p>
          <button
            class="py-1 cursor-pointer bg-slate-700"
          >
            Sign out
          </button>
          <a
            class="w-fit py-1 px-2 bg-slate-700"
            href="/articles/create-article"
          >
            Create Article
          </a>
        </div>
      </div>
      <div
        class="flex items-center mx-4 bg-slate-800"
      >
        <h1>
          Feed
        </h1>
      </div>
      <ul
        class="flex flex-col mx-4 [&_li]:not-last:mb-10 bg-slate-700"
      >
        <li
          class="w-fit hover:underline"
          id="5d6da210-1710-4c6b-b777-7c6e0c29080f"
        >
          <a
            href="articles/5d6da210-1710-4c6b-b777-7c6e0c29080f"
          >
            <h2>
              Acsi enim comparo unde labore strues defessus quisquam clam absque.
            </h2>
          </a>
        </li>
        <li
          class="w-fit hover:underline"
          id="daa7057b-7720-4c5f-a358-c7158ac208c0"
        >
          <a
            href="articles/daa7057b-7720-4c5f-a358-c7158ac208c0"
          >
            <h2>
              Cotidie acquiro amor sperno thermae.
            </h2>
          </a>
        </li>
        <li
          class="w-fit hover:underline"
          id="b3ac39ba-94bf-46c0-8d36-41e587231845"
        >
          <a
            href="articles/b3ac39ba-94bf-46c0-8d36-41e587231845"
          >
            <h2>
              Corrupti tepesco tandem placeat cuius subiungo abduco terga vel.
            </h2>
          </a>
        </li>
        <li
          class="w-fit hover:underline"
          id="a26e9c87-d19f-483d-94b8-f9d0f15f0dd7"
        >
          <a
            href="articles/a26e9c87-d19f-483d-94b8-f9d0f15f0dd7"
          >
            <h2>
              Voluptates comitatus vomito repellat corrupti umquam quaerat.
            </h2>
          </a>
        </li>
        <li
          class="w-fit hover:underline"
          id="50ab9a22-8e27-4672-b54d-c69284daa17f"
        >
          <a
            href="articles/50ab9a22-8e27-4672-b54d-c69284daa17f"
          >
            <h2>
              Consequatur calco damno crebro delinquo volva copia teres.
            </h2>
          </a>
        </li>
        <li
          class="w-fit hover:underline"
          id="216c3c43-066a-4120-a1a6-58fdbe0175e7"
        >
          <a
            href="articles/216c3c43-066a-4120-a1a6-58fdbe0175e7"
          >
            <h2>
              Tempus delinquo bene adhuc comminor adfero suasoria spargo curvo.
            </h2>
          </a>
        </li>
        <li
          class="w-fit hover:underline"
          id="34d61e3d-e798-49d4-8e91-39322d9f5999"
        >
          <a
            href="articles/34d61e3d-e798-49d4-8e91-39322d9f5999"
          >
            <h2>
              Celer cogo corrigo caries nulla subito aequitas volup.
            </h2>
          </a>
        </li>
        <li
          class="w-fit hover:underline"
          id="4de5ae1e-3da5-42ad-adaf-9365f6f82151"
        >
          <a
            href="articles/4de5ae1e-3da5-42ad-adaf-9365f6f82151"
          >
            <h2>
              Derelinquo convoco cetera vesper crepusculum depromo caterva vae crebro.
            </h2>
          </a>
        </li>
      </ul>
    </section>
  </main>
</DocumentFragment>
`;
